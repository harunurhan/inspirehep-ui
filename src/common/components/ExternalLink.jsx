import React from 'react';
import PropTypes from 'prop-types';

function ExternalLink(props) {
  const { children, href, ...otherProps } = props;
  return (
    <a target="_blank" rel="noopener noreferrer" href={href} {...otherProps}>
      {children}
    </a>
  );
}

ExternalLink.propTypes = {
  href: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

export default ExternalLink;
