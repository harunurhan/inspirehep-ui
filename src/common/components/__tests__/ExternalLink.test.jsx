import React from 'react';
import { shallow } from 'enzyme';

import ExternalLink from '../ExternalLink';

describe('ExternalLink', () => {
  it('renders with only href and children', () => {
    const wrapper = shallow(
      <ExternalLink href="//example.com">example</ExternalLink>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with only href, children and extra props', () => {
    const wrapper = shallow(
      <ExternalLink href="//example.com" className="test">
        example
      </ExternalLink>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
