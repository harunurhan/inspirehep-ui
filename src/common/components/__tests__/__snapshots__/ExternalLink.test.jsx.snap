// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalLink renders with only href and children 1`] = `
<a
  href="//example.com"
  rel="noopener noreferrer"
  target="_blank"
>
  example
</a>
`;

exports[`ExternalLink renders with only href, children and extra props 1`] = `
<a
  className="test"
  href="//example.com"
  rel="noopener noreferrer"
  target="_blank"
>
  example
</a>
`;
